core/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/AnnotationTargetProcessor.java

```diff
@@ -430,7 +464,7 @@ boolean fieldSpecifiesAnnotation(Schema fieldSchema) {
             Schema::getDeprecated,
             Schema::getDescription,
             Schema::getExample,
-            Schema::getExtensions,
+            //            Schema::getExtensions,
```

Comment:
Should this be reverted?

Reply:
I think it should be, but uncommenting this line alone causes test failures in the TCK because `getExtensions()` doesn't currently behave like other getter methods. In particular it will never return `null` and the code that uses this list of getters is explicitly checking for whether they return `null`.

To fix this, I had to update the extensions methods on `SchemaImpl` to behave like other getter/setter sets and differentiate between being `null` and being empty. [`ModelConstructionTest.processExtensible`](https://github.com/eclipse/microprofile-open-api/blob/972a655efc638079c88e2491076d0c179e48230a/tck/src/main/java/org/eclipse/microprofile/openapi/tck/ModelConstructionTest.java#L1750) from the TCK also expects this behaviour, but the exact way it was testing it wasn't tripped up by our current implementation.